Marlin/Configuration_adv.h

```diff
@@ -2071,6 +2071,28 @@
   //#define FILAMENT_LCD_DISPLAY
 #endif
 
+/**
+ * Power Monitor
+ * Monitor the PSU voltaqe and/or total system current drawn
+ *
+ * The current sensor feeds a DC voltage (relative to the current drawn) into one of the analog CPU inputs
+ * The voltage sensor feeds a DC voltage (relative to the PSU voltage supply) into one of the analog CPU inputs
+ */
+//#define POWER_MONITOR_CURRENT   // Monitor the system current
+//#define POWER_MONITOR_VOLTAGE   // Monitor the system voltage
+#if EITHER(POWER_MONITOR_CURRENT, POWER_MONITOR_VOLTAGE)
+  //#define POWER_MONITOR_POWER                   // Display calculated W=I*V with both sensor inputs activated
+
+  #define POWER_MONITOR_VOLTS_PER_AMP   0.05000   // Input voltage to the MCU analog pin per amp  - DO NOT exceed the ADC reference voltage!
+  #define POWER_MONITOR_VOLTS_PER_VOLT  0.11786   // Input voltage to the MCU analog pin per volt - DO NOT exceed the ADC reference voltage!
+
+  // ADC reference voltage appropriate to the board. Suggested values:
+  //   3.0 : Possible ADC reference voltage on 32-bit MCUs
+  //   3.3 : Typical ADC reference voltage on 32-bit MCUs
+  //   5.0 : Typical ADC reference voltage on 8-bit MCUs
+  #define POWER_MONITOR_ADC_VREF 5.0
```

Comment:
If this is not set, then it could be set to a default based on the MCU platform and/or board. As far as I know we don't yet have a define that tells us the ADC reference voltage on every board. That would be a good thing to have, if only for reference.

Reply:
mmm sounds good!

Reply:
Maybe have **#define ADC_VREF x.x** in every pin file ?

Reply:
Maybe. We need to see if this is supplied by the platforms.